src/components/schema/PropertiesViewer.tsx

```diff
@@ -47,15 +55,28 @@ function PropertiesViewer({ disabled, editorProps }: Props) {
                     <FormattedMessage id="schemaEditor.fields.label" />
                 </Typography>
 
-                {disabled ? (
-                    <Box style={{ width: 150 }}>
-                        <ExistFilter
-                            fieldFilter={fieldFilter}
-                            setFieldFilter={setFieldFilter}
-                            disabled={inferSchemaResponseEmpty}
+                <Stack
+                    direction="row"
+                    spacing={1}
+                    sx={{ alignItems: 'center' }}
+                >
+                    {disabled ? (
+                        <Box style={{ width: 150 }}>
+                            <ExistFilter
+                                fieldFilter={fieldFilter}
+                                setFieldFilter={setFieldFilter}
+                                disabled={inferSchemaResponseEmpty}
+                            />
+                        </Box>
+                    ) : null}
+
+                    {isCaptureWorkflow ? (
+                        <TableColumnSelector
+                            optionalColumns={optionalColumns}
+                            tablePrefix={TablePrefixes.schemaViewer}
```

Comment:
We should probably hide this when the collection is in edit mode

![image](https://github.com/user-attachments/assets/7c64bd59-0a08-4ec9-a674-98e9e7b41078)